### PR TITLE
Allow to collapse jobs at `/targets` page

### PR DESCRIPTION
**Pull request by hagen1778**
#2417 

It allows to collapse jobs at `/targets` page with saving state at `localStorage`. Also add alert-icon to job's   `th` if there is an alert with endpoints inside
f2228ec215eb42bd8bf610a67a82a1e8

@juliusv

### CODE REVIEW

**Comment by Gouthamve** [comment]
This would be a good feature to have, thanks @hagen1778! Sorry for not taking a look earlier.

I just ran this on my local and it works like a charm!

@juliusv Would be able to review this?

**Comment by Gouthamve** [comment]
Also, please run `make assets` and then push the changed bindata file once you are satisfied.

**Comment by hagen1778** [comment]
Hi! I am using localStorage to keep the "open/close" state of targets. But it looks like using of it is `unsafe` and, probably, this feature would be rejected

**Comment by fabxc** [comment]
Mh, unsafe in how far? Obviously storing passwords in it is not a good
idea. But caching the users choice of collapsing seems perfectly fine.

**Comment by hagen1778** [comment]
It was just my assumption based in `codacy` review. If you deem it acceptable - just great. Also I am ready to implement it in some other way if it's needed.

**Comment by fabxc** [comment]
Tried it out myself and everything works well. Two things came to mind:

* Can we turn the `isHealthy` functions into `numUnhealthy` and return the number of unhealthy targets. Then we can add `(all targets up)` and `({{ .numUnhealthy}} targets down)` next to the job name. That might be a bit clearer than just the warning icon.
* Can we have some indiciation that things are collpsable, like an arrow head maybe? Currently I can click the whole title bar, but that's non-obvious if one doesn't know this exists.

@juliusv @beorn7 comments on having this feature in general?

**Comment by juliusv** [comment]
It would be good to keep this consistent with the collapsing and health indication code we have in https://github.com/prometheus/prometheus/blob/master/web/ui/static/js/alerts.js for alerts. Either update that one or the other one to match. I think the glyphicon exclamation mark looks a bit cartoony for some reason:

exlc

I think I prefer the simple red background of the /alerts page.

> Can we turn the isHealthy functions into numUnhealthy and return the number of unhealthy targets. Then we can add (all targets up) and ({{ .numUnhealthy}} targets down) next to the job name. That might be a bit clearer than just the warning icon.

I'd propose just always showing "(n / m up)", like "(5 / 10 up)".

> Can we have some indiciation that things are collpsable, like an arrow head maybe? Currently I can click the whole title bar, but that's non-obvious if one doesn't know this exists.

It seems that `/alerts` had this and the HTML still includes those indicators (`i.icon-chevron-down`...), but someone must have removed FontAwesome at some point. We should fix that too.

**Comment by fabxc** [comment]
> I'd propose just always showing "(n / m up)", like "(5 / 10 up)".

That makes more sense, yes!

**Comment by hagen1778** [comment]
Good points! That's how I should have done from the very beginning.

**Comment by hagen1778** [comment]
There is error in ci/circleci check:
`Error deleting container: Error response from daemon: Driver btrfs failed to remove root filesystem c69aa24c9e25636a91df341cdb3810699f3982f9cd62405f0a967edadd339478: Failed to destroy btrfs snapshot /var/lib/docker/btrfs/subvolumes for c69aa24c9e25636a91df341cdb3810699f3982f9cd62405f0a967edadd339478: operation not permitted`

I'm not sure that can handle it

**Comment by beorn7** [comment]
The circleci test was just a temporary failure of quay.io (I believe). Worked now.

Is everybody fine with this PR?

**Comment by fabxc** [comment]
Fine with me.

File: web/ui/templates/targets.html [hunk 1]
```diff
@@ -1,57 +1,72 @@
-{{define "head"}}<!-- nix -->{{end}}
+{{define "head"}}
+<link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/css/targets.css?v={{ buildVersion }}">
```
Comment (grobie) [inline web/ui/templates/targets.html]: What's the reason to break this out into another css file? The overhead of another HTTP roundtrip to fetch the file should be higher than to the few bytes added to the file even when not looking at the targets page.

Comment (hagen1778) [inline web/ui/templates/targets.html]: just to be consistent with alerts.html and alerts.css. Should it be in the one file?

Comment (grobie) [inline web/ui/templates/targets.html]: I see, thanks for the info. Not sure the overhead is worth it, but let's keep it consistent for now.

**Comment by beorn7** [comment]
I sense a consensus of approval. I'll merge this.

**Comment by knweiss** [comment]
As mentioned in #3050 I wonder how `web/ui/bindata.go` was created for this PR because it references local paths (`/Users/hagen/PhpstormProjects/go/src/github.com/hagen1778/...`). Was `DEBUG=1` set while executing `make assets`?

**Comment by beorn7** [comment]
Thanks @knweiss . I have created #3051 to fix.

**Comment by hagen1778** [comment]
My apologies for broken `bindata.go`

**Pull request merged**